Formula/s/spglib.rb

```diff
@@ -0,0 +1,66 @@
+class Spglib < Formula
+  desc "C library for finding and handling crystal symmetries"
+  homepage "https://spglib.readthedocs.io/"
+  url "https://github.com/spglib/spglib/archive/refs/tags/v2.1.0.tar.gz"
+  sha256 "31bca273a1bc54e1cff4058eebe7c0a35d5f9b489579e84667d8e005c73dcc13"
+  license "BSD-3-Clause"
+
+  depends_on "cmake" => [:build, :test]
+  depends_on "gcc" # for gfortran
+
+  def install
+    common_args = %w[
+      -DSPGLIB_WITH_Fortran=ON
+    ]
+    [true, false].each do |shared_lib_type|
+      build_dir = "build_#{shared_lib_type ? "shared" : "static"}"
+      system "cmake",
+             "-S", ".",
+             "-B", build_dir,
+             *common_args,
+             "-DSPGLIB_SHARED_LIBS=#{shared_lib_type ? "ON" : "OFF"}",
+             *std_cmake_args
+      system "cmake", "--build", build_dir
+      system "ctest",
+             "-j", ENV.make_jobs,
+             "--test-dir", build_dir,
+             "--output-on-failure"
```

Comment:
We don't run tests in packages.

Reply:
Ok, I've seen a few of them that run. Any advice on how to run these upstream or equivalent, maybe by maintaining an upstream tap? It does pick up on some errors that we should address upstream.

Reply:
Run it in upstream CI?

Reply:
Yes, if possible to setup an upstream tap to run the same GitHub actions as here, but with upstream testing enabled, that would be ideal. I haven't looked into how to set it up, but from another discussion I understood that would be possible.